Server - rework how the Pi modname is generated/used

samaaron opened the pull request:
The pi's modname is currently used for 2 things:

1. To determine the default sched ahead time (depending on whether it's pi 2, pi 3 or something later).
2. To generate the user-readable host name displayed in the bottom right of the GUI.

We now use a smarter way of determining the info for the Pi (code submitted by Robin Newman) rather than relying on an ever-growing list of codes. This is then used to generate the readable host name from which we can easily determine if it's a pi2, pi3 or something else.

samaaron committed:
Server - rework how the Pi modname is generated/used

The pi's modname is currently used for 2 things:

1. To determine the default sched ahead time (depending on whether it's pi 2, pi 3 or something later).
2. To generate the user-readable host name displayed in the bottom right of the GUI.

We now use a smarter way of determining the info for the Pi (code submitted by Robin Newman) rather than relying on an ever-growing list of codes. This is then used to generate the readable host name from which we can easily determine if it's a pi2, pi3 or something else.

rbnpi commented:
Hmm dropped your version in and it is not producing a message on my Pi4 screen. Mine is checked giving correct messages on Pi3 Pi3+ Pi4 and Pi400. Pi2 is not supported on latest OS. All builds will be 64bit no tailwind  32bit available, Not had a chance to check why yours is not producing the messages.
My version is working OK. I just wasn't sure about whether I needed the variables to be @@ or not.

Have a look at my PR submitted yesterday for this and the daemon file.

samaaron commented:
Mine is untested as I don't have access to a pi right now. I think I prefer cleaning things up like I have done. So would be useful to have a play with it to see why it's not working if you get a moment. I'll take a look to see if I can see the issue too. Is everything else working though?

samaaron committed:
Server - update method of fetching pi modname from shell

samaaron commented:
Just updated based on your other pr. I wonder if this helps things.

rbnpi commented:
Still not working. The problem is you have not defined all the types used in `def host_platform_desc`
The selection bit above seems to work OK  as detect_pimodel returns "Raspberry Pi 4B:4Gb 64bit OS" when I copy the code to my pi and run it but needs to be expanded to give all the definitions required in def host_platform_desc which gives errors.

samaaron committed:
Server - remove older function definitoin

This was duplicated for reference when developing the new approach but accidentally not removed.

samaaron commented:
oh hahaha, I think I see the issue. I duplicated the function so I could have a copy of the original to reference but then failed to delete it! 

This might work now...

rbnpi commented:
Looks OK now. Tried on Pi4 4gb and 8gb and also on pi3 and pi3b + they seem ok.
